src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs

```diff
@@ -128,10 +129,12 @@ private void AddOrRemoveDropdown()
                     return;
                 }
 
-                // Temporary solution until the editor provides a proper way to resolve the correct navbar.
-                // Tracked in https://github.com/dotnet/roslyn/issues/40989
-                var document = _languageService.EditorAdaptersFactoryService.GetDataBuffer(buffer)?.AsTextContainer().GetRelatedDocuments().FirstOrDefault();
-                if (document?.GetLanguageService<INavigationBarItemService>() == null)
+                var textBuffer = _languageService.EditorAdaptersFactoryService.GetDataBuffer(buffer);
+                var document = textBuffer.AsTextContainer().GetRelatedDocuments().FirstOrDefault();
+                // TODO - Remove the TS check once they move the liveshare navbar to LSP.  Then we can also switch to LSP
+                // for the local navbar implementation.
+                // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163360
+                if (textBuffer != null && textBuffer.IsInCloudEnvironmentClientContext() && document.Project.Language != "TypeScript")
```

Comment:
If `textBuffer` was null, line 133 would have already thrown

Reply:
We already have the document for a workspace, any reason to be doing the lookup a second time with the textBuffer.IsInCloudEnvironmentClientContext() or should we just ask the workspace service from the document we already have?